webapp/src/containers/LinkModal/index.tsx

```diff
@@ -1,138 +1,126 @@
-        setLinkPayload(payload);
-
         // Make POST api request
-        usePlugin.makeApiRequest(Constants.pluginApiServiceConfigs.createLink.apiServiceName, payload);
-    }, [state]);
+        linkTask(projectDetails);
+    };
 
-    useEffect(() => {
-        if (linkPayload) {
-            const {isLoading, isSuccess, isError} = usePlugin.getApiState(Constants.pluginApiServiceConfigs.createLink.apiServiceName, linkPayload);
-            setLoading(isLoading);
-            if (isSuccess) {
-                onHide();
-            }
-            if (isError) {
-                setAPIError('Organization or project name is wrong');
-            }
+    // Make POST api request to link a project
+    const linkTask = async (payload: LinkPayload) => {
+        const createTaskRequest = await usePlugin.makeApiRequest(plugin_constants.pluginApiServiceConfigs.createLink.apiServiceName, payload);
+        if (createTaskRequest) {
+            dispatch(toggleIsLinked(true));
+            resetModalState();
         }
-    }, [usePlugin.state]);
-
-    if (visibility) {
-        return (
-            <Modal
-                show={visibility}
-                title='Link new project'
-                onHide={onHide}
-                onConfirm={onConfirm}
-                confirmBtnText='Link new project'
-                cancelDisabled={loading}
-                confirmDisabled={loading}
-                loading={loading}
-                error={APIError}
-            >
-                <>
-                    <Input
-                        type='text'
-                        placeholder='Organization name'
-                        value={state.linkOrganization}
-                        onChange={onOrganizationChange}
-                        error={error.linkOrganizationError}
-                        required={true}
-                    />
-                    <Input
-                        type='text'
-                        placeholder='Project name'
-                        value={state.linkProject}
-                        onChange={onProjectChange}
-                        required={true}
-                        error={error.linkProjectError}
-                    />
-                </>
-            </Modal>
-        );
-    }
-    return null;
+    };
+
+    useEffect(() => {
+        setProjectDetails({
+            organization: getLinkModalState(usePlugin.state).organization,
+            project: getLinkModalState(usePlugin.state).project,
+        });
+    }, [getLinkModalState(usePlugin.state)]);
+
+    return (
+        <Modal
+            show={getLinkModalState(usePlugin.state).visibility}
+            title='Link new project'
+            onHide={resetModalState}
+            onConfirm={onConfirm}
+            confirmBtnText='Link new project'
+            cancelDisabled={usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.createLink.apiServiceName, projectDetails).isLoading}
```

Comment:
Extract all these variables like `isLoading`, `isSuccess` etc. above the `return` block.